server/Routes/Consumer/Consumer.py

```diff
@@ -20,21 +20,30 @@ def check_auth(self):
         if data['token'] != Configuration.CONSUMER_TOKEN:
             return abort(HTTPStatus.UNAUTHORIZED)
 
-    @User_auth.login_required
     def post(self):
         parser = reqparse.RequestParser()
         parser.add_argument('team1_id', type=str, required=True)
         parser.add_argument('team2_id', type=str, required=True)
         parser.add_argument('name', type=str, required=True)
         data = parser.parse_args()
+        match_file = request.files['file']
+        if not match_file.filename.endswith('.zip'):
+            return make_response(jsonify({'message': 'File must be a zip file'}), 400)
         with EventsManager(data['name']) as eventmanager:
             eventdata = eventmanager.find_event()
             data['event_id'] = eventdata['_id']
+            blob_storage = BlobStorageModel()
+            container = blob_storage.get_container(str(eventdata['_id']))
+            if not container:
+                container = blob_storage.create_container(
+                    str(eventdata['_id']))
             with TeamsManager() as teamsmanager:
                 challenging_team = teamsmanager.find_team_by_id(ObjectId(data['team1_id']))
                 if challenging_team['last_challenge_timestamp'] == None:
                     with MatchRequestManager() as requestmanager:
```

Comment:
The match result should be posted with a zip, seems this is posting for the request post?

Reply:
Indeed, this post request shouldn't need to handle and zip files. Post the result zip to /api/match

Reply:
I think consumer upload the zip so it should be done within post request right?

Reply:
@KyrelJerome When a match is done (i.e. done the request), the id of the request will be the id of the match result (so you have to provide the match_id when create POST match). Then you can access the file with match_<match_id>